pkg/registry/generic/registry/storage_factory.go

```diff
@@ -30,7 +30,8 @@ func StorageWithCacher(
 	objectType runtime.Object,
 	resourcePrefix string,
 	scopeStrategy rest.NamespaceScopedStrategy,
-	newListFunc func() runtime.Object) storage.Interface {
+	newListFunc func() runtime.Object,
+	indices ...storage.IndexingFunc) storage.Interface {
```

Comment:
I don't like passing in the indices here. It will force knowledge of indexing to be spread around too many places in the code base. I would prefer a system that automatically adds them.